### ISSUE TITLE
SteamCMD app_info_print no output to stdout before quit - Recent Issue

### ISSUE THREAD

**Issue by own3mall** [ValveSoftware/steam-for-linux#9683]
#### Your system information

Latest steamcmd linux script:

`./steamcmd.sh +login anonymous +app_info_update 1 +app_info_print 232130 +quit`

#### Please describe your issue in as much detail as possible:
`./steamcmd.sh +login anonymous +app_info_update 1 +app_info_print 232130 +quit` produces no output before exit.

`./steamcmd.sh +login anonymous +app_info_update 1 +app_info_print 232130` prints output but doesn't exit (because no `quit` parameter passed in)

#### Steps for reproducing this issue:

1. Run this command:  `./steamcmd.sh +login anonymous +app_info_update 1 +app_info_print 232130 +quit`
2. No app_info_print output.

It should output to stdout the same thing it does when running the script without the `+quit` parameter:

`./steamcmd.sh +login anonymous +app_info_update 1 +app_info_print 232130`

**Comment by tbeswick96** [comment]
Am having the same problem on Windows.
Same command but with APP ID 233780 (don't think that matters)

Same thing working without `+quit` bt not working with `+quit`. Sometimes it dumps the data, most of the time it does not.

**Comment by HunterZ** [comment]
For me it sometimes prints the info and sometimes does not. Very strange.

**Comment by GGHostDotGames** [comment]
Same problem here with StreamCMD on Windows. It started about two weeks ago. I ended up putting the command in a loop, which checks if the required output is there and, if not, repeats the command. Loops every 0.5s and usually returns the results within ten iterations.

**Comment by GGHostDotGames** [comment]
Based on the output it does give on failed attempts, it would seem as if it fails on executions where SteamCMD decides to autoupdate itself. The successful executions seem to be the ones where the auto-update was skipped.

For example, a successful execution's output:

```
Steam Console Client (c) Valve Corporation - version 1687387651
-- type 'quit' to exit --
Loading Steam API...OK
src\public\tier1\interface.h (265) : m_pModule
AppID : 4940, change number : 19358343/0, last change : Fri Jun 30 21:02:11 2023 
"4940"
{
	"common"
	{
	
	etc, etc
```
	
Vs a failed one:
```
Steam Console Client (c) Valve Corporation - version 1687387651
-- type 'quit' to exit --
Loading Steam API...OK
src\public\tier1\interface.h (265) : m_pModule
Redirecting stderr to 'C:\Users\juanh\Desktop\SCUMUpdater4\steamcmd\logs\stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
```

**Comment by 0x00000ED** [comment]
Windows.

Hello, I’ll say right away that everything worked before, but now I can’t get the vdf package in node js
I run node script like this

let cmd = spawn(
this.#launcher,
['+login', this.#username, '+app_info_update', '0', '+app_info_print', serverAppID, "+quit"],
{cwd: this.#folder, shell: false}
);

cmd.stdout.on('data', (data) => {
stdout += data.toString();

console.log(data.toString());
});


// this.#username = "anonymous"
// serverAppID = "232330"

stamcmd exits and I don't get information about the requested application

log
```
Steam Console Client (c) Valve Corporation - version 1691628584
-- type 'quit' to exit --
Loading Steam API...OK

Connecting anonymously to Steam Public...
OK
Waiting for client config...OK
Waiting for user info...
OK

Redirecting stderr to 'D:\server\server\[SteamCmd]\logs\stderr.txt'
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[] Проверка на наличие обновлений...
[----] Проверка установки...
```

How to fix it?
please don't suggest to use runscript

**Comment by ThoronadNg** [comment]
still the same problem here.
when i use steamcmd.exe +login anonymous +app_info_update 1 +app_info_print 232130 +quit 
it rarly prints out the app_info before quit.

**Comment by Sipowicz** [comment]
Hello everyone,

TL:DR - I use this and it works now:
`steamcmd.exe +login anonymous +app_info_update +app_info_print steamAppID +logoff +quit`

I am working with Windows server 2022, but the issue was the same when using SteamCMD

`steamcmd.exe +login anonymous +app_info_update 1 +app_info_print $steamAppID +app_info_print $steamAppID +quit` - normally produces the wrong output, but randomly produces the correct output.

When I originally started using this command in 2016 it worked fine. 
After few years it randomly stopped working reliably which I was able to fix by adding a few more "+app_info_print"s
`steamcmd.exe +login anonymous +app_info_update 1 +app_info_print $steamAppID +app_info_print $steamAppID +app_info_print $steamAppID +app_info_print $steamAppID +quit`  - This worked up until a few weeks ago (as far as I can tell).

It seemed like the data pull was being cut off by the **_+quit_** options, so I added **_+logoff_** before **_+quit_**, and that seems to have done the trick.

**Comment by tbeswick96** [comment]
So doing the +logoff worked to make it dump the output. Now I occasionally get invalid JSON:
Why is there an `OK` in the middle of the dump? Seems possibly like text dumps are getting mixed up as that OK is from a previous message (e.g `Logging off current session...<OK should be here>`).
![](https://i.imgur.com/L9gpSiq.png)

**Comment by brandonsturgeon** [comment]
I found something that works for me:

```
steamcmd.sh \
    +login anonymous \
    +app_info_request <id>\
    +logoff \
    +login anonymous \
    +app_info_print <id> \
    +quit
```

If we're lucky, `+logoff` and `+login` actually wait for all pending requests to finish.

If we're unlucky, this is just introducing an artificial delay which gives `app_info_request` enough time to finish.

In the event that the latter is true, we could just keep adding more `+logoff +login anonymous` until you get a sufficiently safe delay... 😅

**Comment by 39sdev** [comment]
> The last solution here stopped working for me (within the last few days).
> 
> However, I found something that works again (for me):
> 
> ```
> steamcmd.sh \
>     +login anonymous \
>     +app_info_request <id> \
>     +login anonymous \
>     +app_info_print <id> \
>     +quit
> ```

this works most of the time. but for some apps it also fails.

I got this response from it:
```
AppID : 730, change number : 26584392/26584392, last change : Sun Dec 15 13:25:00 2024 
"730"
{
}
```

and it just stays that way, even if rerun.
is there some cache it keeps?

**Comment by own3mall** [comment]
They broke it:  

https://github.com/ValveSoftware/steam-for-linux/issues/11521

**Comment by FlaminSarge** [comment]
Was `app_info_print` made non-blocking for some reason? Made to mirror the behavior of app_info_request plus print as a side-effect instead of as an end result?

**Comment by svanegmond** [comment]
Please fix this absurd bug. The info command should return a response before moving to the next command.

**Comment by JulianStiebler** [comment]
I found working to login with a unlimited steam account in a interactive steamCMD, as opposed to programatically; sending app_info_print may still result in a empty response like "No app info found, requesting...", but sending it a second time does bring back a full response.

Doing it programatically (even with unlimited steam account) (sending twice after a detected empty response) just returns a empty dictionary :/
